base/string.jl

```diff
@@ -968,8 +968,8 @@ byte_string_classify(s::ByteString) = byte_string_classify(s.data)
     # 1: valid ASCII
     # 2: valid UTF-8
 
-is_valid_ascii(s::Union(Array{UInt8,1},ByteString)) = byte_string_classify(s) == 1
-is_valid_utf8(s::Union(Array{UInt8,1},ByteString)) = byte_string_classify(s) != 0
+isvalid(::Type{ASCIIString}, s::Union(Array{UInt8,1},ByteString)) = byte_string_classify(s) == 1
```

Comment:
Unfortunately, if I can't manage to get the big (10x) performance issues with Unicode conversions in, I may have trouble keeping my clients interested in using Julia... in which case I won't be having that much time to improve Julia further, much as I'd like to... I've made all the changes that people have requested in #11004 (except for ones that ran into Julia bugs), now just waiting...